File: app/src/main/java/org/triplepy/sh8email/sh8/di/module/ClientModule.java
package org.triplepy.sh8email.sh8.di.module;

import org.triplepy.sh8email.sh8.api.Sh8Client;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

/**
 * The sh8email-android Project.
 * ==============================
 * org.triplepy.sh8email.sh8.di.module
 * ==============================
 * Created by igangsan on 2016. 8. 28..
 */

@Module
public class ClientModule {
    @Provides
    public OkHttpClient provideOkHttpClient() {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
        return new OkHttpClient.Builder()
                .addInterceptor(logging)
                .build();
    }

    @Provides
    public Sh8Client provideSh8Client(){
        return new Sh8Client(provideOkHttpClient());
    }
}

File: app/src/main/kotlin/org/triplepy/sh8email/sh8/di/module/ClientModule.kt
package org.triplepy.sh8email.sh8.di.module

import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import org.triplepy.sh8email.sh8.api.Sh8Client

/**
 * The sh8email-android Project.
 * ==============================
 * org.triplepy.sh8email.sh8.di.module
 * ==============================
 * Created by igangsan on 2016. 8. 28..
 */

@Module
class ClientModule {
    @Provides
    fun provideOkHttpClient(): OkHttpClient {
        val logging = HttpLoggingInterceptor()
        logging.level = HttpLoggingInterceptor.Level.BODY
        return OkHttpClient.Builder().addInterceptor(logging).build()
    }

    @Provides
    fun provideSh8Client(): Sh8Client {
        return Sh8Client(provideOkHttpClient())
    }
}
